pennylane/ops/qubit/attributes.py

```diff
@@ -199,3 +199,29 @@ def __contains__(self, obj):
 representation using ``np.linalg.eigvals``, which fails for some tensor types that the matrix
 may be cast in on backpropagation devices.
 """
+
+supports_tensorbatching = Attribute(
```

Comment:
I may be entirely wrong, but this is something that to me actually feels like part of the class, rather than an attribute!

To me, attributes are physical properties of the operation (`diagonal_in_z_basis`, `symmetric_over_control_wires`, etc.) that are mostly unrelated to code structure.

But `supports_tensorbatching` feels more akin to a logical property of the operation, like `num_params` and `num_wires`?

Reply:
What do you think? (@mariaschuld might also have some thoughts)

Reply:
Josh is right, so far we have used attributes more for maths properties. But that's not a given! 

The idea of ``Attributes`` is to collate information that ops provide for a specific application (like a compilation or batching) to one point. If the application gets deleted, the ``Attribute`` gets deleted. To "register" a new op with the application, one has to add it to the list (which is obviously a bit opaque, but the same is true if the dev has to know somehow that there is a class attribute/flag inside some operator classes).

I'd say that the way you do it is the best one. We could think of "functional" and "mathematical" attributes going forward and put them into different files, but for now this is clear logic, and allows future changes to be easily implemented.

Reply:
Alright, thanks a lot :)